src/lib.rs

```diff
@@ -43,43 +55,112 @@ impl fmt::Debug for Shim {
     }
 }
 
-/// Initializes the panic hook.
+/// Determines how backtraces will be displayed.
+#[cfg(feature = "with-backtrace")]
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum BacktraceMode {
+    /// Backtraces will be omitted from the log.
+    Off,
+    /// Backtraces will include addresses, but no symbol names or locations.
+    Unresolved,
+    /// Backtraces will include addresses as well as symbol names and locations when possible.
+    Resolved
+}
+
+/// Configures the panic hook, ending with initialization.
 ///
-/// After this method is called, all panics will be logged rather than printed
-/// to standard error.
-pub fn init() {
-    panic::set_hook(Box::new(|info| {
-        let backtrace = Backtrace::default();
-
-        let thread = thread::current();
-        let thread = thread.name().unwrap_or("<unnamed>");
-
-        let msg = match info.payload().downcast_ref::<&'static str>() {
-            Some(s) => *s,
-            None => match info.payload().downcast_ref::<String>() {
-                Some(s) => &**s,
-                None => "Box<Any>",
-            },
+/// ## Example
+///
+/// ```
+/// # #[cfg(feature = "with-backtrace")]
+/// log_panics::Config::new()
+///     .backtrace_mode(log_panics::BacktraceMode::Unresolved)
+///     .install_panic_hook()
+/// ```
+#[derive(Debug)]
+pub struct Config {
+    // We store a constructor function instead of a BacktraceMode enum
+    // so that inlining can eliminate references to `Backtrace::default`
+    // if symbolication is not desired.
+    make_backtrace: fn() -> Backtrace,
+}
+
+impl Config {
+    /// Initializes the builder with the default set of features.
+    pub fn new() -> Self {
+        Self {
+            make_backtrace: Backtrace::default,
+        }
+    }
+
+    /// Controls how backtraces are displayed.
+    ///
+    /// The default when backtraces are enabled is [`BacktraceMode::Resolved`].
+    #[cfg(feature = "with-backtrace")]
+    pub fn backtrace_mode(mut self, mode: BacktraceMode) -> Self {
+        self.make_backtrace = match mode {
+            BacktraceMode::Off => || Backtrace::from(vec![]),
+            BacktraceMode::Unresolved => Backtrace::new_unresolved,
+            BacktraceMode::Resolved => Backtrace::default,
         };
+        self
+    }
+
+    /// Initializes the panic hook.
+    ///
+    /// After this method is called, all panics will be logged rather than printed
+    /// to standard error.
+    #[inline]
```

Comment:
This inline doesn't seem necessary.

Reply:
Oops, yeah, that's left over from previous testing of not resolving the enum eagerly. Will remove.